File: app/src/main/java/hottopic/mit/co/nz/cleaningservice/network/DiscountsRequest.java
package hottopic.mit.co.nz.cleaningservice.network;

import android.content.Context;

import java.util.Map;

import hottopic.mit.co.nz.cleaningservice.entities.network.DiscountsResponse;
import hottopic.mit.co.nz.cleaningservice.network.service.PaymentService;
import io.reactivex.Single;

public class DiscountsRequest extends BaseRequest<PaymentService, DiscountsResponse, DiscountsResponse> {

    public DiscountsRequest(Context context) {
        super(context);
    }

    @Override
    protected Map<String, String> getHeader() {
        return null;
    }

    @Override
    protected Class<PaymentService> getEndpointClass() {
        return PaymentService.class;
    }

    @Override
    protected Single<DiscountsResponse> getEndpoint(PaymentService endpoint) {
        return endpoint.getDiscouts();
    }

    @Override
    protected DiscountsResponse dealWithResponse(DiscountsResponse response) {
        return response;
    }
}

File: app/src/main/java/hottopic/mit/co/nz/cleaningservice/network/DiscountsRequest.kt
package hottopic.mit.co.nz.cleaningservice.network

import android.content.Context

import hottopic.mit.co.nz.cleaningservice.entities.network.response.DiscountsResponse
import hottopic.mit.co.nz.cleaningservice.network.service.PaymentService
import io.reactivex.Single

class DiscountsRequest(context: Context) : BaseRequest<PaymentService, DiscountsResponse, DiscountsResponse>(context) {

    override val header: Map<String, String>?
        get() = null

    override val endpointClass: Class<PaymentService>
        get() = PaymentService::class.java

    override fun getEndpoint(endpoint: PaymentService?): Single<DiscountsResponse?>? {
        return endpoint?.discouts
    }

    override fun dealWithResponse(response: DiscountsResponse): DiscountsResponse? {
        return response
    }
}
